crates/adapters/src/integrated/delta_table/input.rs

```diff
@@ -273,23 +289,72 @@ impl DeltaTableInputEndpointInner {
         }
     }
 
+    /// Compute the subset of columns in the Delta table schema that occur in the SQL
+    /// table declaration.
+    fn relevant_columns(&self, table: &DeltaTable) -> Vec<String> {
+        // Column names in the SQL schema.
+        let sql_columns = self
+            .schema
+            .fields
+            .iter()
+            .map(|field| field.name.name())
+            .collect::<Vec<String>>();
+
+        if let Some(table_schema) = table.schema() {
+            let delta_columns = table_schema
+                .fields()
+                .map(|f| f.name().clone())
+                .collect::<Vec<String>>();
+
+            // We need to be careful in checking whether a Delta column name occurs in
+            // sql_columns.  Delta doesn't seem to include case sensitivity information any
+            // any form in its schema.  So we conservatively check whether column name
+            // occurs in the SQL tables as is or whether a lowercase column name occurs
+            // in the set of SQL columns converted to lowercase.
+
+            let sql_columns = sql_columns.iter().cloned().collect::<BTreeSet<_>>();
+            let sql_columns_lowercase = sql_columns
+                .iter()
+                .map(|c| c.to_lowercase())
+                .collect::<BTreeSet<_>>();
+
+            delta_columns
+                .into_iter()
+                .filter(|c| {
+                    sql_columns.contains(c) || sql_columns_lowercase.contains(&c.to_lowercase())
```

Comment:
doesn't the second test imply the first one?

Reply:
Usually yes, but I think `to_lowercase` can do funny things for some non-ASCII characters, so the first test may still be necessary in some cases.